internal/dependencyinjection/main.go

```diff
@@ -1,9 +1,27 @@
 package dependencyinjection
 
-// dependenciesStoreMap var is the global map for all the service instances.
-var dependenciesStoreMap map[string]interface{} = map[string]interface{}{}
+import "sync"
+
+var (
+	// dependenciesStoreMap var is the global map for all the service instances.
+	dependenciesStoreMap = make(map[string]interface{})
+	// dependenciesStoreMutex is the mutex for the global map, used to make sure the map is thread safe.
+	dependenciesStoreMutex sync.Mutex
```

Comment:
Why don't we use the [`sync.Map`](https://pkg.go.dev/sync#Map) type here? Only a suggestion, no need to change it.

Reply:
Goof idea, done in https://github.com/stellar/stellar-disbursement-platform-backend/pull/165/commits/d1289c36522c84aea1d84f898cc9b679bf5b2c47.